src/loki/search.cc

```diff
@@ -697,6 +709,12 @@ struct bin_handler_t {
     // need to go get the actual correlated location with edge_id etc.
     std::unordered_map<Location, PathLocation> searched;
     for (auto& pp : pps) {
+      // remove non-sensical island candidates
+      auto new_end = std::remove_if(pp.unreachable.begin(), pp.unreachable.end(),
+                                    [&pp](const candidate_t& c) -> bool {
+                                      return c.sq_distance > pp.closest_external_reachable;
+                                    });
+      pp.unreachable.erase(new_end, pp.unreachable.end());
```

Comment:
this is where we remove those island candidates that we should have never kept in the first place